docs: refactor (WIP) (#840)
aidenybai/million: website/pages/docs/block.mdx

import { Callout } from 'nextra-theme-docs';
import { AutomaticModeWarning } from '../../components/automatic-mode-warning';

> ## Quick Run Of A Block?

Author: @LadyBluenotes , this is 2

> A block is a specialized [Higher Order Component (HOC)](https://legacy.reactjs.org/docs/higher-order-components.html) that functions as a React component but more optimized for rendering speed.

Author: @LadyBluenotes, this is 1

> Blocks are components wrapped by `block(){:jsx}`.

Author: Hmm, thanks for this!

import { Callout } from 'nextra-theme-docs';
import { AutomaticModeWarning } from '../../components/automatic-mode-warning';

> ## What's a block?

Reviewer: Re: this page

You may want to determine how you want the structure of the reference docs to look like.

I'm not sure if you've finished this page yet or not, but there's a bit of repetition within it. It's also a bit different than how `<For>` is.

Author: Thanks for this, I resolved to this.

<img width="1241" alt="Screenshot 2023-12-07 at 13 39 47" src="https://github.com/aidenybai/million/assets/96334363/df0af49f-5d8d-4bf4-b3f7-082f900f1334">

> Million.js is a library that enables you to create _blocks_. A block is a special [Higher Order Component (HOC)](https://legacy.reactjs.org/docs/higher-order-components.html) that can be used as a React component but is hyper-optimized for rendering speed.

Reviewer: The first sentence here is created lower down. If this is a reference doc, then you don't have to talk about how Million lets you create blocks.

```suggestion
A block is a specialized [Higher Order Component (HOC)](https://legacy.reactjs.org/docs/higher-order-components.html) that functions as a React component but more optimized for rendering speed.
```

Author: Thanks for this again, Sarah! I really appreciate it.

> Million.js is a library that enables you to create _blocks_. A block is a special [Higher Order Component (HOC)](https://legacy.reactjs.org/docs/higher-order-components.html) that can be used as a React component but is hyper-optimized for rendering speed.

Author: Thank you very, much for this! 🙌

> Blocks are essentially components wrapped by `block(){:jsx}`.

Reviewer: I try to avoid "can" or words of the like if things are concrete. So if blocks *are* just a component wrapped by `block(){:jsx}` then you can say that, unless there are edge cases. In which case, you should elabroate those.

```suggestion
Blocks are components wrapped by `block(){:jsx}`.
```

Author: Thanks, Sarah!!! 🙌!! You're so experienced. Damn!!

  Blocks](/docs/rules) for more information.
</Callout>

> ## Custom Tags

Author: Thank you!!

);
```

> ## Using Block With SSR

Author: I really appreciate that.
